examples/seismic/model.py

```diff
@@ -50,61 +50,72 @@ class Model(object):
     :param m: The square slowness of the wave
     :param damp: The damping field for absorbing boundarycondition
     """
-    def __init__(self, origin, spacing, vp, nbpml=0, dtype=np.float32,
-                 rho=None, epsilon=None, delta=None, theta=None, phi=None):
+    def __init__(self, origin, spacing, shape, vp, nbpml=0, dtype=np.float32,
+                 epsilon=None, delta=None, theta=None, phi=None):
         self.vp = vp
         self.origin = origin
         self.spacing = spacing
         self.nbpml = nbpml
         self.dtype = dtype
+        self.shape = shape
         # Create square slowness of the wave as symbol `m`
-        self.m = DenseData(name="m", shape=self.shape_domain, dtype=self.dtype)
-        self.m.data[:] = self.pad(1 / (self.vp * self.vp))
+        if isinstance(vp, np.ndarray):
+            self.m = DenseData(name="m", shape=self.shape_domain, dtype=self.dtype)
+            self.m.data[:] = self.pad(1 / (self.vp * self.vp))
+        else:
+            self.m = 1/vp**2
```

Comment:
Yes it is, I guess `vp` can be dropped, needs some changes in the functions accessing it.